modules/databend/databend.go

```diff
@@ -0,0 +1,146 @@
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	databendUser        = "databend"
+	defaultUser         = "databend"
+	defaultPassword     = "databend"
+	defaultDatabaseName = "default"
+)
+
+// DatabendContainer represents the Databend container type used in the module
+type DatabendContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+var _ testcontainers.ContainerCustomizer = (*DatabendOption)(nil)
+
+// DatabendOption is an option for the Databend container.
+type DatabendOption func(*DatabendContainer)
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o DatabendOption) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// Run creates an instance of the Databend container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DatabendContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{"8000/tcp"},
+		Env: map[string]string{
+			"QUERY_DEFAULT_USER":     defaultUser,
+			"QUERY_DEFAULT_PASSWORD": defaultPassword,
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog("port: 8000  Databend Query Server"),
+			wait.ForListeningPort("8000/tcp"),
+		),
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, err
+		}
+	}
+
+	username := req.Env["QUERY_DEFAULT_USER"]
+	password := req.Env["QUERY_DEFAULT_PASSWORD"]
+	if password == "" && username == "" {
+		return nil, errors.New("empty password and user")
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *DatabendContainer
+	if container != nil {
+		c = &DatabendContainer{
+			Container: container,
+			password:  password,
+			username:  username,
+			database:  defaultDatabaseName,
+		}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("generic container: %w", err)
+	}
+
+	return c, nil
+}
+
+// MustConnectionString panics if the address cannot be determined.
+func (c *DatabendContainer) MustConnectionString(ctx context.Context, args ...string) string {
+	addr, err := c.ConnectionString(ctx, args...)
+	if err != nil {
+		panic(err)
+	}
+	return addr
+}
+
+func (c *DatabendContainer) ConnectionString(ctx context.Context, args ...string) (string, error) {
+	containerPort, err := c.MappedPort(ctx, "8000/tcp")
+	if err != nil {
+		return "", err
+	}
+
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	extraArgs := ""
+	if len(args) > 0 {
+		extraArgs = strings.Join(args, "&")
+	}
+	if extraArgs != "" {
+		extraArgs = "?" + extraArgs
+	}
+	if c.database == "" {
+		c.database = defaultDatabaseName
+	}
```

Comment:
User can use `c.ConnectionString` to switch different database which is the same as extra arg?

Reply:
My understanding is that the user would use ConnectionString to enable them to connect to the container however they might have more than one database so passing that as a parameter into this method would enable that to work. Otherwise it they would have to create a new container or manually manipulate the result to do that.

It would be simple to have this method look for a database parameter in extra params and override the default to make that flow easy.

Does that make sense?

Reply:
I think user can use `db.execute("use newDatabase")` to switch which is more convenient.  If add it in extra params, user also need to create a new connection string to connect.